OptimoveSDK/Sources/Classes/Optimobile/InApp/InAppPresenter.swift

```diff
@@ -130,38 +160,31 @@ class InAppPresenter : NSObject, WKScriptMessageHandler, WKNavigationDelegate{
         }
         
         messageQueueLock.wait()
-        defer {
-            messageQueueLock.signal()
-        }
         
         messageQueue.removeObject(at: 0)
         pendingTickleIds.remove(message.id)
         currentMessage = nil
         
         if messageQueue.count == 0 {
             pendingTickleIds.removeAllObjects()
-            self.destroyViews()
-        }
-        else {
-            presentFromQueue()
         }
+
+        messageQueueLock.signal()
```

Comment:
The lock is now also acquired inside presentFromQueue() so needs to be released prior to entering that scope.